src/omv/ports/stm32/modules/py_display.c

```diff
@@ -0,0 +1,649 @@
+/*
+ * This file is part of the OpenMV project.
+ *
+ * Copyright (c) 2013-2021 Ibrahim Abdelkader <iabdalkader@openmv.io>
+ * Copyright (c) 2013-2021 Kwabena W. Agyeman <kwagyeman@openmv.io>
+ *
+ * This work is licensed under the MIT license, see the file LICENSE for details.
+ *
+ * LTDC/DSI display Python module.
+ */
+#include "omv_boardconfig.h"
+
+#if MICROPY_PY_DISPLAY &&                   \
+    (defined(OMV_RGB_DISPLAY_CONTROLLER) || \
+    defined(OMV_DSI_DISPLAY_CONTROLLER))
+
+#include "py/obj.h"
+#include "py/runtime.h"
+#include "mphal.h"
+
+#include "py_helper.h"
+#include "py_image.h"
+#include "omv_gpio.h"
+#include "py_display.h"
+
+#if defined(OMV_DSI_DISPLAY_BL_PIN)
+#define OMV_DISPLAY_BL_PIN OMV_DSI_DISPLAY_BL_PIN
+#elif defined(OMV_RGB_DISPLAY_BL_PIN)
+#define OMV_DISPLAY_BL_PIN OMV_RGB_DISPLAY_BL_PIN
+#endif
+
+#if defined(OMV_DSI_DISPLAY_DISP_PIN)
+#define OMV_DISPLAY_DISP_PIN OMV_DSI_DISPLAY_DISP_PIN
+#elif defined(OMV_RGB_DISPLAY_DISP_PIN)
+#define OMV_DISPLAY_DISP_PIN OMV_RGB_DISPLAY_DISP_PIN
+#endif
+
+typedef struct display_mode {
+    uint32_t hactive;
+    uint32_t vactive;
+    uint32_t pixel_clock;
+    uint32_t hsync_len;
+    uint32_t hback_porch;
+    uint32_t hfront_porch;
+    uint32_t vsync_len;
+    uint32_t vback_porch;
+    uint32_t vfront_porch;
+    uint32_t hpol : 1;
```

Comment:
Did you pass the polarities along? They don't seem to be in the array below.

Reply:
They're all low, so I may just remove those fields.